content/kb/troubleshooting-volume-readonly-or-io-error.md

```diff
@@ -0,0 +1,73 @@
+```
+2m26s       Warning   DetachedUnexpectly       volume/pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c               Engine of volume pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c dead unexpectedly, reattach the volume
+```
+
+When checking for logs of the longhorn manager pods on the node the workload is running by running `kubectl -n longhorn-system logs <longhorn manager pod name> | grep <volume name>`, the following message is shown:
+
+```
+time="2021-01-05T11:20:46Z" level=debug msg="Instance handler updated instance pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c-e-0fe2dac3 state, old state running, new state error"
+time="2021-01-05T11:20:46Z" level=warning msg="Instance pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c-e-0fe2dac3 crashed on Instance Manager instance-manager-e-a1fd54e4 at shuo-cluster-0-worker-3, try to get log"
+......
+time="2021-01-05T11:20:46Z" level=warning msg="Engine of volume dead unexpectedly, reattach the volume" accessMode=rwo controller=longhorn-volume frontend=blockdev node=shuo-cluster-0-worker-3 owner=shuo-cluster-0-worker-3 state=attached volume=pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c
+......
+time="2021-01-05T11:20:46Z" level=info msg="Event(v1.ObjectReference{Kind:\"Volume\", Namespace:\"longhorn-system\", Name:\"pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c\", UID:\"69bb0f94-da48-4d15-b861-add435f25d00\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"7466467\", FieldPath:\"\"}): type: 'Warning' reason: 'DetachedUnexpectly' Engine of volume pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c dead unexpectedly, reattach the volume"
+```
+
+### Most common/Direct reason
+The mount point of the Longhorn volume becomes invalid once the Longhorn volume crashes unexpectedly. Then there is no way to read or write data in the Longhorn volume via the mount point.
+
+For Longhorn version earlier than v1.1.0, Longhorn will try to remount the volume automatically but it doesn't work well.
+
+Since Longhorn version v1.1.0, a new setting [`Automatically Delete Workload Pod when The Volume Is Detached Unexpectedly`](https://longhorn.io/docs/1.1.0/references/settings/#automatically-delete-workload-pod-when-the-volume-is-detached-unexpectedly) is introduced so that Longhorn will automatically delete the workload pod that is managed by a controller (e.g. deployment, statefulset, daemonset, etc...).
+
+#### Solution
+If the workload is a simple pod, you can delete and re-deploy the pod. Please make sure the related PVC or PV is not removed if the reclaim policy is not `Retain`. Otherwise, the Longhorn volume will be removed once the related PVC/PV is gone.
+
+If the workload pod belongs to Deployment/StatefulSet, you can restart the pod by scaling down then scaling up the workload replica.
+
+And for Longhorn v1.1.0 or higher version, you can enable setting `Automatically Delete Workload Pod when The Volume Is Detached Unexpectedly`.
+
+### Root cause
+The key point is, what leads to the unexpected engine crash. Here are the possible reasons/checklist:
```

Comment:
`Engine crashed is normally contributed to it lost the connections to every single replica. Here are the possible reasons why that's happened:`.